src/ProposalTypesConfigurator.sol

```diff
@@ -164,12 +164,59 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
         if (!_proposalTypesExists[proposalTypeId]) revert InvalidProposalType();
 
         if (!scopeExists[proposalTypeId][txTypeHash]) revert InvalidScope();
-        Scope[] memory validScopes = scopes[proposalTypeId];
+        Scope memory validScope = scopes[proposalTypeId][txTypeHash];
+        return validScope.encodedLimits;
+    }
+
+    function getParameter(bytes calldata limit, uint256 startIdx, uint256 endIdx)
+        public
+        pure
+        returns (bytes memory parameter)
+    {
+        return limit[startIdx:endIdx + 1];
+    }
+
+    function validateProposedTx(bytes calldata proposedTx, uint8 proposalTypeId, bytes32 txTypeHash)
+        public
+        view
+        returns (bool valid)
+    {
+        Scope memory validScope = scopes[proposalTypeId][txTypeHash];
+        bytes memory scopeLimit = validScope.encodedLimits;
+        bytes4 selector = bytes4(scopeLimit);
+        if (selector != bytes4(proposedTx[:4])) revert Invalid4ByteSelector();
+
+        uint256 startIdx = 4;
+        uint256 endIdx = 0;
+        for (uint8 i = 0; i < validScope.parameters.length; i++) {
+            if (i == 0) {
+                endIdx = startIdx + validScope.parameters[i].length;
+            } else {
+                endIdx = endIdx + validScope.parameters[i].length;
+            }
+
+            bytes32 param = bytes32(proposedTx[startIdx:endIdx]);
+
+            if (validScope.comparators[i] == Comparators.EQUAL) {
+                bytes32 scopedParam = bytes32(this.getParameter(scopeLimit, startIdx, endIdx));
```

Comment:
it would be nice to have some natspec comments in the interface as well, it's hard to understand the differencen between `encodedLimits` and `parameters`

Reply:
We should decide what's the source of truth here first, the parameters object or the encoded limit. I was going to verify that both compute to the same value but was having trouble finding a robust way of doing so